pkg/operator/configobserver/apiserver/observe_audit.go

```diff
@@ -0,0 +1,88 @@
+package apiserver
+
+import (
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// AuditPolicyPathGetter allows the observer to be agnostic of the source of audit profile(s).
+type AuditPolicyPathGetter interface {
+	// AuditPolicyPathGetter returns the path to the audit policy file (associated with the
+	// given profile) in the static manifest folder.
+	GetAuditPolicyPath(profile string) (string, error)
+
+	// GetDefaultAuditPolicyPath returns the path to the default audit policy.
+	GetDefaultAuditPolicyPath() string
```

Comment:
why isn't that `GetAuditPolicyPath("Default")`?

Reply:
to reduce coupling, library-go doesn't need to know the defaulting mechanism, it just needs to know:
- the path to the default audit profle
- given an audit profile, it needs to know the desired path.
on the other hand, the apiserver/component that defines the audit profiles should know well which audit profile is default.

I know we are defaulting at the CRD level, but i think the observer implementation should still guard for empty profile name and other error conditions.

Reply:
This func is only called to default the value in the old config. This whole block of code is only needed for the event. Just compare to empty string. Same result.

Reply:
It resorts to the default value if the existing/current config is not set. when the existing config is not set and the observer function encounters a lister error or any other before it can set the new/desired configuration, what should the observer return?